tests/unit/test_resultset.py

```diff
@@ -41,6 +41,19 @@ def test_iter_paged(self):
         type(response_future).has_more_pages = PropertyMock(side_effect=(True, True, False))  # after init to avoid side effects being consumed by init
         self.assertListEqual(list(itr), expected)
 
+    def test_iter_paged_with_empty_pages(self):
+        expected = list(range(10))
+        response_future = Mock(has_more_pages=True, _continuous_paging_session=None)
```

Comment:
Why is it ok to always return has_more_pages==True? Shouldn't it return False for the last page?

Reply:
because `stopIteration` will be raised anyway, so in the scope of this test, we do not need to add the complexity of chaging the paging state of the response future to demonstrate and test the fixed behavior

Reply:
I must have a bad day because I don't see how it works.
If `has_more_pages` is `True` then the condition in this [`if`](https://github.com/scylladb/python-driver/blob/55a21e394c08521ce6d6a07107209839439bc09d/cassandra/cluster.py#L5128) will be false and the exception won't be raised, no?
That in turn will lead to `next` being called again recursively over and over gain, no?

or will the `StopIteration` be thrown from [`fetch_next_page`](https://github.com/scylladb/python-driver/blob/55a21e394c08521ce6d6a07107209839439bc09d/cassandra/cluster.py#L5134)?

Reply:
I understand it's not obvious sorry.

The `StopIteration` comes from the mocked response_future.result() iterator over the configured `side_effect`s.

In other words, the tests do iterate over the `side_effect`s every time `response_future.result()` is called in `fetch_next_page()`, when all the `side_effect`s (pages) have been depleted, the next call to `response_future.result()` will raise a `StopIteration` exception.

Does that make more sense?

Reply:
yep. That's what I meant by `StopIteration` being thrown from `fetch_next_page`. Thanks.